modules/munit/src/main/scala/doobie/munit/analysisspec.scala

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2013-2020 Rob Norris and Contributors
+// This software is licensed under the MIT License (MIT).
+// For more information see LICENSE or https://opensource.org/licenses/MIT
+
+package doobie.munit
+
+import cats.effect.{ Effect, IO }
+import doobie.util.query.{Query, Query0}
+import doobie.util.testing._
+import org.tpolecat.typename._
+import munit.Assertions
+import munit.Location
+
+/**
+ * Module with a mix-in trait for specifications that enables checking of doobie `Query` and `Update` values.
+ * 
+ * {{{
+ * class ExampleSuite extends FunSuite with IOChecker {
+ *
+ *   // The transactor to use for the tests.
+ *   val transactor = Transactor.fromDriverManager[IO](
+ *     "org.postgresql.Driver",
+ *     "jdbc:postgresql:world",
+ *     "postgres", ""
+ *   )
+ *
+ *   // Now just mention the queries. Arguments are not used.
+ *   test("findByNameAndAge") { check(MyDaoModule.findByNameAndAge(null, 0)) }
+ *   test("allWoozles") { check(MyDaoModule.allWoozles) }
+ *
+ * }
+ * }}}
+ */
+object analysisspec {
+
+  trait Checker[M[_]] extends CheckerBase[M] { this: Assertions =>
+
+    def check[A: Analyzable](a: A)(implicit loc: Location) = checkImpl(Analyzable.unpack(a))
+
+    def checkOutput[A: TypeName](q: Query0[A])(implicit loc: Location) =
+      checkImpl(AnalysisArgs(
+        s"Query0[${typeName[A]}]", q.pos, q.sql, q.outputAnalysis
+      ))
+
+    def checkOutput[A: TypeName, B: TypeName](q: Query[A, B])(implicit loc: Location) =
+      checkImpl(AnalysisArgs(
+        s"Query[${typeName[A]}, ${typeName[B]}]", q.pos, q.sql, q.outputAnalysis
+      ))
+
+    private def checkImpl(args: AnalysisArgs)(implicit loc: Location) = {
+      val report = analyzeIO(args, transactor).unsafeRunSync()
+      if (!report.succeeded) {
+        fail(
+          formatReport(args, report, colors)
+            .padLeft("  ")
+            .toString
+        )
+      }
```

Comment:
Wouldn't https://github.com/typelevel/munit-cats-effect be better instead?
```suggestion
      val report = analyzeIO(args, transactor)
      report.flatMap{
          case r if !r.succeeded => 
            IO(fail(
              formatReport(args, report, colors)
                .padLeft("  ")
                .toString
            ))
          case _ => IO.unit
       }
```

Reply:
I have implemented the munit-cats-effect integration at 
https://github.com/ValdemarGr/doobie/tree/munit
Can someone give any insight or consideration for the idea?

Reply:
Sounds like a good idea. It'll also make CE3 migration easier

Reply:
This implementation also changes the semantics of check; it is no longer eager, is this desired?
The scalatest and specs2 test utilities seem to be eager.

Reply:
If execution thread gets to `check(..)` then we're running the test already right? (This PR's impl seems eager to me, unless I'm missing something)

munit (and all other test frameworks that I know) all use call-by-name to allow skipping tests, so `test("asdf") { check(..) }` is still skippable even if `check` is eager.

Reply:
I mean the evaluation of the check effect.
In the initially proposed solution check runs eagerly which fails the following test:
https://github.com/ValdemarGr/doobie/blob/6ad3f9414bc7cb6f1a1bbe1e65789315bc38dd08/modules/munit/src/test/scala/doobie/munit/CheckerTests.scala#L31-L34
In the version I published the test combinator is wrapped here:
https://github.com/typelevel/munit-cats-effect/blob/b0845550c337a21d72c46f9e2619a02a4479ad88/common/shared/src/main/scala/munit/CatsEffectFunFixtures.scala#L68-L79
Thus the test succeeds in the munit-cats-effect variant:
https://github.com/ValdemarGr/doobie/blob/ebbd0ccc475bf21dd5ebee4967741c4ff0263992/modules/munit/src/test/scala/doobie/munit/CheckerTests.scala#L33-L36

Reply:
Ah yeah I think the other integrations (scalatest/specs) run it eagerly. Perhaps it is worth keeping that behaviour to make it easier for people to migrate between test libs. (I know that in my codebase I have a few `check(..)` being called as if they're statements)

So I guess cats-effect munit won't help us here. Sorry!